src/main/java/redis/clients/jedis/Jedis.java

```diff
@@ -8297,6 +8297,22 @@ public String clusterMyId() {
             .map(BuilderFactory.ENCODED_OBJECT_MAP::build).collect(Collectors.toList());
   }
 
+  @Override
+  public String clusterAddSlotsRange(int... ranges) {
```

Comment:
I don't think it's a good idea to use `int... ranges` as args here. Because the start and end slots are required. My suggestion is 
You can introduce new class `SlotRange` like the below code.
```
class SlotRange {
   int startSlot;
   int endSlot;
}
```
then you can do in this way  clusterAddSlotsRange(SlotRange... slotRanges) {}

Reply:
@dengliming I think this is okay. Many users prefer this way. Also, we have similar pattern in other commands, e. g. MSET.

We can add your suggestion alongside these methods.

Reply:
You were right. There is no doubt that this is the easiest way. but users must avoid passing only one parameter. e.g. 
clusterAddSlotsRange(1). Or we can check the params. `if (paths.length % 2 != 0) { throw err }`

Reply:
You're right but Jedis doesn't do that, at least not yet :laughing: